src/model/test/HeatPumpPlantLoopEIRCooling_GTest.cpp

```diff
@@ -0,0 +1,157 @@
+  EXPECT_EQ(1.0, hp.sizingFactor());
+  boost::optional<CurveBiquadratic> capacityModifierFunctionofTemperatureCurve = hp.capacityModifierFunctionofTemperatureCurve();
+  EXPECT_TRUE(capacityModifierFunctionofTemperatureCurve);
+  boost::optional<CurveBiquadratic> electricInputtoOutputRatioModifierFunctionofTemperatureCurve =
+    hp.electricInputtoOutputRatioModifierFunctionofTemperatureCurve();
+  EXPECT_TRUE(electricInputtoOutputRatioModifierFunctionofTemperatureCurve);
+  boost::optional<CurveQuadratic> electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve =
+    hp.electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve();
+  EXPECT_TRUE(electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve);
+}
+
+TEST_F(ModelFixture, HeatPumpPlantLoopEIRCooling_GettersSetters) {
+  Model m;
+  HeatPumpPlantLoopEIRCooling hp(m);
+
+  EXPECT_TRUE(hp.setCondenserType("AirSource"));
+  HeatPumpPlantLoopEIRHeating companionHP(m);
+  EXPECT_TRUE(hp.setCompanionHeatingHeatPump(companionHP));
+  EXPECT_TRUE(hp.setReferenceLoadSideFlowRate(1.0));
+  EXPECT_TRUE(hp.setReferenceSourceSideFlowRate(2.0));
+  EXPECT_TRUE(hp.setReferenceCapacity(3.0));
+  EXPECT_TRUE(hp.setReferenceCoefficientofPerformance(4.0));
+  EXPECT_TRUE(hp.setSizingFactor(5.0));
+  CurveBiquadratic curve1(m);
+  EXPECT_TRUE(hp.setCapacityModifierFunctionofTemperatureCurve(curve1));
+  CurveBiquadratic curve2(m);
+  EXPECT_TRUE(hp.setElectricInputtoOutputRatioModifierFunctionofTemperatureCurve(curve2));
+  CurveQuadratic curve3(m);
+  EXPECT_TRUE(hp.setElectricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve(curve3));
+
+  EXPECT_EQ("AirSource", hp.condenserType());
+  ASSERT_TRUE(hp.companionHeatingHeatPump());
+  EXPECT_EQ(companionHP.handle(), hp.companionHeatingHeatPump().get().handle());
+  ASSERT_TRUE(hp.referenceLoadSideFlowRate());
+  EXPECT_EQ(1.0, hp.referenceLoadSideFlowRate().get());
+  EXPECT_FALSE(hp.isReferenceLoadSideFlowRateAutosized());
+  ASSERT_TRUE(hp.referenceSourceSideFlowRate());
+  EXPECT_EQ(2.0, hp.referenceSourceSideFlowRate());
+  EXPECT_FALSE(hp.isReferenceSourceSideFlowRateAutosized());
+  ASSERT_TRUE(hp.referenceCapacity());
+  EXPECT_EQ(3.0, hp.referenceCapacity().get());
+  EXPECT_FALSE(hp.isReferenceCapacityAutosized());
+  EXPECT_EQ(4.0, hp.referenceCoefficientofPerformance());
+  EXPECT_EQ(5.0, hp.sizingFactor());
+  EXPECT_EQ(curve1.handle(), hp.capacityModifierFunctionofTemperatureCurve().handle());
+  EXPECT_EQ(curve2.handle(), hp.electricInputtoOutputRatioModifierFunctionofTemperatureCurve().handle());
+  EXPECT_EQ(curve3.handle(), hp.electricInputtoOutputRatioModifierFunctionofPartLoadRatioCurve().handle());
+
+  hp.autosizeReferenceLoadSideFlowRate();
+  hp.autosizeReferenceSourceSideFlowRate();
+  hp.autosizeReferenceCapacity();
+
+  EXPECT_TRUE(hp.isReferenceLoadSideFlowRateAutosized());
+  EXPECT_TRUE(hp.isReferenceSourceSideFlowRateAutosized());
+  EXPECT_TRUE(hp.isReferenceCapacityAutosized());
```

Comment:
By deferring this at the end instead of fully testing each field individually you're not actually testing that your reset functions are reseting the right field. I expect they are since the implementation (model .cpp) likely has been done via GenerateClass.rb, but that's beside the point here.

GenerateClass.rb should generate a model test too no?